types/record.d.ts

```diff
@@ -31,6 +33,8 @@ declare class Record {
 
   forEach(visitor: Visitor): void
 
+  map(visitor: MapVisitor): Array<any>
```

Comment:
Would be nice to add a test for this new function declaration in `record.test.ts`